Index: trunk/Source/WebCore/dom/Document.cpp
===================================================================
--- trunk/Source/WebCore/dom/Document.cpp	(revision 104838)
+++ trunk/Source/WebCore/dom/Document.cpp	(revision 104845)
@@ -2982,6 +2982,8 @@

void Document::styleSelectorChanged(StyleSelectorUpdateFlag updateFlag)
{
     // Don't bother updating, since we haven't loaded all our style info yet
     // and haven't calculated the style selector for the first time.
-    if (!attached() || (!m_didCalculateStyleSelector && !haveStylesheetsLoaded()))
-        return;
+    if (!attached() || (!m_didCalculateStyleSelector && !haveStylesheetsLoaded())) {
+        m_styleSelector.clear();
+        return;
+    }
 
 #ifdef INSTRUMENT_LAYOUT_SCHEDULING
    if (!ownerElement())
        printf("Beginning update of style selector at time %d.\n", elapsedTime());
#endif
